packages/firestore/src/core/firestore_client.ts

```diff
@@ -297,6 +300,20 @@ export class FirestoreClient {
     return true;
   }
 
+  /**
+   * Checks that the client has not been shutdown. Ensures that other methods on
+   * this class cannot be called after the client is shutdown.
+   */
+  private checkClientRunning(): void {
+    if (this.isClientShutdown === false) {
+      return;
+    }
+    throw new FirestoreError(
+      Code.FAILED_PRECONDITION,
+      'The client has already been shutdown. Please restart the client.'
```

Comment:
There's not really a way to "restart" the client...  You'd need to get a new instance of `Firestore` by calling `firebase.firestore()` again.  Perhaps just drop "Please restart the client."